sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2826,8 +2826,18 @@ public Write<T> withJsonTimePartitioning(ValueProvider<String> partitioning) {
      * tables, the fields here will be ignored; call {@link #withClustering()} instead.
      */
     public Write<T> withClustering(Clustering clustering) {
+      return withJsonClustering(StaticValueProvider.of(BigQueryHelpers.toJsonString(clustering)));
+    }
+
+    /** Like {@link #withClustering(Clustering)} but using a deferred {@link ValueProvider}. */
+    public Write<T> withClustering(ValueProvider<Clustering> clustering) {
+      return withJsonClustering(NestedValueProvider.of(clustering, BigQueryHelpers::toJsonString));
+    }
+
+    /** The same as {@link #withClustering(Clustering)}, but takes a JSON-serialized object. */
```

Comment:
Can we give an example as to how the string should be formatted and what fields should be there ?

Reply:
Thanks, added